auction-house/program/src/receipt/mod.rs

```diff
@@ -0,0 +1,507 @@
+            LISTING_RECEIPT_PREFIX.as_bytes(),
+            seller_trade_state.pubkey.as_ref(),
+            &[receipt_bump],
+        ];
+
+        create_or_allocate_account_raw(
+            *ctx.program_id,
+            &receipt_info,
+            &rent.to_account_info(),
+            &system_program,
+            &bookkeeper_account,
+            LISTING_RECEIPT_SIZE,
+            &[],
+            &receipt_seeds,
+        )?;
+    }
+
+    let receipt = ListingReceipt {
+        trade_state: seller_trade_state.pubkey,
+        bookkeeper: bookkeeper_account.key(),
+        auction_house: auction_house.pubkey,
+        seller: wallet.pubkey,
+        metadata: metadata.pubkey,
+        purchase_receipt: None,
+        price: sell_data.buyer_price,
+        token_size: sell_data.token_size,
+        bump: receipt_bump,
+        trade_state_bump: sell_data.trade_state_bump,
+        created_at: clock.unix_timestamp,
+        canceled_at: None,
+    };
+
+    receipt.try_serialize(&mut *receipt_account.try_borrow_mut_data()?)?;
+
+    Ok(())
+}
+
+#[derive(Accounts)]
+pub struct CancelListingReceipt<'info> {
+    #[account(mut)]
+    receipt: UncheckedAccount<'info>,
+    system_program: Program<'info, System>,
+    instruction: UncheckedAccount<'info>,
+}
+
+pub fn cancel_listing_receipt<'info>(
+    ctx: Context<'_, '_, '_, 'info, CancelListingReceipt<'info>>,
+) -> ProgramResult {
+    let receipt_account = &ctx.accounts.receipt;
+    let instruction_account = &ctx.accounts.instruction;
+    let clock = Clock::get()?;
+
+    let receipt_info = receipt_account.to_account_info();
+
+    let prev_instruction = get_instruction_relative(-1, instruction_account)?;
+    let prev_instruction_accounts = prev_instruction.accounts;
+
+    let trade_state = &prev_instruction_accounts[6];
+
+    assert_program_instruction_equal(
+        &prev_instruction.data[..8],
+        [232, 219, 223, 41, 219, 236, 220, 190],
+    )?;
+
+    if receipt_info.data_is_empty() {
+        return Err(ErrorCode::ReceiptIsEmpty.into());
+    }
+
+    assert_derivation(
+        &id(),
+        &receipt_info,
+        &[LISTING_RECEIPT_PREFIX.as_ref(), trade_state.pubkey.as_ref()],
+    )?;
+
+    let receipt_data = receipt_info.try_borrow_mut_data()?;
+    let mut receipt_data_slice: &[u8] = &receipt_data;
+
+    let mut receipt = ListingReceipt::try_deserialize(&mut receipt_data_slice)?;
+
+    receipt.canceled_at = Some(clock.unix_timestamp);
+
+    let mut data = receipt_data;
+    let dst: &mut [u8] = &mut data;
+    let mut cursor = std::io::Cursor::new(dst);
+
+    receipt.try_serialize(&mut cursor)?;
+
+    Ok(())
+}
+
+#[derive(Accounts)]
+#[instruction(receipt_bump: u8)]
+pub struct PrintBidReceipt<'info> {
+    #[account(mut)]
+    receipt: UncheckedAccount<'info>,
+    #[account(mut)]
+    bookkeeper: Signer<'info>,
+    system_program: Program<'info, System>,
+    rent: Sysvar<'info, Rent>,
+    instruction: UncheckedAccount<'info>,
```

Comment:
No Sysvar is available for instructions so passing the program id as an unchecked account. Let me know if there is an alternative.

Reply:
https://github.com/metaplex-foundation/metaplex-program-library/pull/181/commits/570736061c93ed266ea743876c9db52705c5e410